src/cbs.c

```diff
@@ -460,9 +465,35 @@ int cbs_close(CBS_HANDLE cbs)
     return result;
 }
 
-int cbs_put_token_async(CBS_HANDLE cbs, const char* type, const char* audience, const char* token, ON_CBS_OPERATION_COMPLETE on_cbs_put_token_complete, void* on_cbs_put_token_complete_context)
+static void cbs_put_token_cancel_handler(ASYNC_OPERATION_HANDLE put_token_operation)
 {
-    int result;
+    CBS_OPERATION* cbs_operation = GET_ASYNC_OPERATION_CONTEXT(CBS_OPERATION, put_token_operation);
+
+    (void)async_operation_cancel(cbs_operation->amqp_management_async_context);
+
+    LIST_ITEM_HANDLE list_node = singlylinkedlist_get_head_item(cbs_operation->pending_operations);
+
+    while (list_node != NULL)
+    {
+        if (singlylinkedlist_item_get_value(list_node) == cbs_operation)
+        {
+            if (singlylinkedlist_remove(cbs_operation->pending_operations, list_node) != 0)
+            {
+                LogError("Failed removing pending operation from list.");
+            }
+
+            async_operation_destroy(cbs_operation->token_operation_async_context);
+
+            break;
+        }
+
+        list_node = singlylinkedlist_get_next_item(list_node);
+    }
```

Comment:
Same here, I remember singlylinkedlist having some delete with a condition ...

Reply:
Yes, I could use that one. `singlylinkedlist_remove_if`

Reply:
Done.